build.gradle.kts

```diff
@@ -0,0 +1,71 @@
+import com.example.buildsrc.MyExtension
+
+plugins {
+    id("java")
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnitPlatform()
+}
+
+allprojects{
+    extensions.create("Extension", MyExtension::class.java)
+
+    val addElementNoProjectAllScope by extra { _: Unit ->
+        configure<MyExtension> {
+            contents.add("addElementNoProjectAllScope")
+        }
+    }
+
+    //passing a no-argument lambda makes the val initialized with lambda result
+    val addElementFakeLambdaAllScope by extra {
+        configure<MyExtension> {
+            contents.add("addElementFakeLambdaAllScope")
+        }
+    }
+}
+
+val addElementNoProjectExtScope by extra { _: Unit ->
+    configure<MyExtension> {
+        contents.add("addElementNoProjectExtScope")
+    }
+}
+
+//passing a no-argument lambda makes the val initialized with lambda result
```

Comment:
Okay... and the evaluation here is, remind me, lazy or eager? pls also state in code (in case this repo leads to a blogpost one day)

Reply:
All those vals are evaluated eagerly, the ones in allprojects scope are initialized in both the root project and the subproject right in the Gradle's _Configure project task_ (which is the first one after tasks builidng buildSrc/).